components/server/src/redis/mutex.ts

```diff
@@ -35,4 +36,25 @@ export class RedisMutex {
             automaticExtensionThreshold: 1 * 1000, // time in ms
         });
     }
+
+    public async using<T>(
+        resources: string[],
+        duration: number,
+        routine: (signal: RedlockAbortSignal) => Promise<T>,
+    ): Promise<T> {
+        try {
+            return this.client().using(resources, duration, routine);
+        } catch (err) {
+            if (err instanceof ExecutionError) {
+                // Workaround for https://github.com/mike-marcacci/node-redlock/issues/168 and https://github.com/mike-marcacci/node-redlock/issues/169
+                if (err.message === "The operation was unable to achieve a quorum during its retry window") {
+                    log.debug("wsgc: failed to acquire workspace-gc lock, another instance already has the lock", err);
+
+                    throw new ResourceLockedError("operation was unable to achieve quorum");
+                }
+            }
+
+            throw err;
+        }
+    }
```

Comment:
The wrapping is done such that we keep the same interface, but we can uniformly catch the execution error